File: src/main/java/dragonsaga/items/containers/RaceChangeContainer.java
package dragonsaga.items.containers;

import dragonsaga.race.DragonRace;
import dragonsaga.registry.DragonRaceRegistry;
import necesse.engine.network.NetworkClient;
import necesse.inventory.container.Container;
import necesse.inventory.container.customAction.BooleanCustomAction;
import necesse.inventory.container.customAction.EmptyCustomAction;
import necesse.inventory.container.customAction.EnumCustomAction;
import necesse.inventory.container.customAction.IntCustomAction;

public class RaceChangeContainer extends Container {

    public EnumCustomAction<DragonRaceRegistry.RaceEnum> playerRaceButton;
    EmptyCustomAction applyRace;

    public RaceChangeContainer(NetworkClient client, int uniqueSeed) {
        super(client, uniqueSeed);

        this.playerRaceButton = this.registerAction(new EnumCustomAction<DragonRaceRegistry.RaceEnum>(DragonRaceRegistry.RaceEnum.class) {
            @Override
            protected void run(DragonRaceRegistry.RaceEnum race) {
                DragonRace.Instance.ChangePlayerRaceNextTick(client.playerMob, race);
            }
        });

    }
}

File: src/main/kotlin/dragonsaga/items/containers/RaceChangeContainer.kt
package dragonsaga.items.containers

import dragonsaga.registry.RaceEnum
import dragonsaga.race.DragonRace
import necesse.engine.network.NetworkClient
import necesse.inventory.container.Container
import necesse.inventory.container.customAction.EmptyCustomAction
import necesse.inventory.container.customAction.EnumCustomAction

class RaceChangeContainer(client: NetworkClient, uniqueSeed: Int) : Container(client, uniqueSeed) {
    var playerRaceButton: EnumCustomAction<RaceEnum>
    var applyRace: EmptyCustomAction? = null

    init {
        playerRaceButton = registerAction(object : EnumCustomAction<RaceEnum>(RaceEnum::class.java) {
            override fun run(race: RaceEnum?) {
                DragonRace.Instance.ChangePlayerRaceNextTick(client.playerMob, race)
            }
        })
    }
}